README.md

```diff
@@ -0,0 +1,84 @@
+# JsonApiDotNetCore MongoDB Repository
+
+[![NuGet version][nuget-image]][nuget-url]
+[![Downloads][downloads-image]][nuget-url]
+[![Build Status](https://travis-ci.com/mrnkr/JsonApiDotNetCore.MongoDb.svg?branch=master)](https://travis-ci.com/mrnkr/JsonApiDotNetCore.MongoDb)
+[![codecov](https://codecov.io/gh/mrnkr/JsonApiDotNetCore.MongoDb/branch/master/graph/badge.svg)](https://codecov.io/gh/mrnkr/JsonApiDotNetCore.MongoDb)
+[![license][license]](https://github.com/mrnkr/JsonApiDotNetCore.MongoDb/blob/master/LICENSE)
+
+[nuget-image]:https://img.shields.io/nuget/v/JsonApiDotNetCore.MongoDb
+[nuget-url]:https://www.nuget.org/packages/JsonApiDotNetCore.MongoDb
+[downloads-image]:https://img.shields.io/nuget/dt/JsonApiDotNetCore.MongoDb
+[license]:https://img.shields.io/github/license/mrnkr/JsonApiDotNetCore.MongoDb
+
+Plug-n-play implementation of `IResourceRepository<TResource, TId>` allowing you to use MongoDb with your `JsonApiDotNetCore` APIs.
+
+## Installation and Usage
+
+### Models
+
+```cs
+public sealed class Book : IIdentifiable<string>
```

Comment:
I tried but it caused an error, let me show you what I did:

```cs
public sealed class Book : Identifiable<string>
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    [Attr]
    public new string Id { get; set; }

    // ... The other props ...
    
    [BsonIgnore]
    public new string StringId { get => Id; set => Id = value; }
}
```

The error this caused is `MongoDB.Bson.BsonSerializationException: The property 'Id' of type 'Example.Models.Book' cannot use element name '_id' because it is already being used by property 'Id' of type 'JsonApiDotNetCore.Resources.Identifiable`1[[System.String, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]'.`.

Overriding the property this way seems to be causing problems for the MongoDb driver to map the fields properly, using the interface, however, this does not happen so I take it as a win but let me know your thoughts on this.

Just for the record, here is the complete error log:

```
fail: JsonApiDotNetCore.Middleware.ExceptionHandler[0]
      The property 'Id' of type 'Example.Models.Book' cannot use element name '_id' because it is already being used by property 'Id' of type 'JsonApiDotNetCore.Resources.Identifiable`1[[System.String, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]'.
MongoDB.Bson.BsonSerializationException: The property 'Id' of type 'Example.Models.Book' cannot use element name '_id' because it is already being used by property 'Id' of type 'JsonApiDotNetCore.Resources.Identifiable`1[[System.String, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]'.
   at BsonClassMap MongoDB.Bson.Serialization.BsonClassMap.Freeze()
   at BsonClassMap MongoDB.Bson.Serialization.BsonClassMap.LookupClassMap(Type classType)
   at IBsonSerializer MongoDB.Bson.Serialization.BsonClassMapSerializationProvider.GetSerializer(Type type, IBsonSerializerRegistry serializerRegistry)
   at IBsonSerializer MongoDB.Bson.Serialization.BsonSerializerRegistry.CreateSerializer(Type type)
   at TValue System.Collections.Concurrent.ConcurrentDictionary<TKey, TValue>.GetOrAdd(TKey key, Func<TKey, TValue> valueFactory)
   at IBsonSerializer MongoDB.Bson.Serialization.BsonSerializerRegistry.GetSerializer(Type type)
   at IBsonSerializer<T> MongoDB.Bson.Serialization.BsonSerializerRegistry.GetSerializer<T>()
   at new MongoDB.Driver.MongoCollectionImpl<TDocument>(IMongoDatabase database, CollectionNamespace collectionNamespace, MongoCollectionSettings settings, ICluster cluster, IOperationExecutor operationExecutor)
   at IMongoCollection<TDocument> MongoDB.Driver.MongoDatabaseImpl.GetCollection<TDocument>(string name, MongoCollectionSettings settings)
   at IMongoCollection<TResource> JsonApiDotNetCore.MongoDb.MongoEntityRepository<TResource, TId>.get_Collection() in /Users/mrnkr/Documents/code/JsonApiDotNetCore.MongoDb/src/JsonApiDotNetCore.MongoDb/MongoEntityRepository.cs:line 39
   at IMongoQueryable<TResource> JsonApiDotNetCore.MongoDb.MongoEntityRepository<TResource, TId>.get_Entities() in /Users/mrnkr/Documents/code/JsonApiDotNetCore.MongoDb/src/JsonApiDotNetCore.MongoDb/MongoEntityRepository.cs:line 40
   at IMongoQueryable<TResource> JsonApiDotNetCore.MongoDb.MongoEntityRepository<TResource, TId>.ApplyQueryLayer(QueryLayer layer) in /Users/mrnkr/Documents/code/JsonApiDotNetCore.MongoDb/src/JsonApiDotNetCore.MongoDb/MongoEntityRepository.cs:line 117
   at Task<int> JsonApiDotNetCore.MongoDb.MongoEntityRepository<TResource, TId>.CountAsync(FilterExpression topFilter, CancellationToken cancellationToken) in /Users/mrnkr/Documents/code/JsonApiDotNetCore.MongoDb/src/JsonApiDotNetCore.MongoDb/MongoEntityRepository.cs:line 53
   at TRet System.Dynamic.UpdateDelegates.UpdateAndExecute3<T0, T1, T2, TRet>(CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at async Task<int> JsonApiDotNetCore.Repositories.ResourceRepositoryAccessor.CountAsync<TResource>(FilterExpression topFilter, CancellationToken cancellationToken) in C:/projects/jsonapidotnetcore/src/JsonApiDotNetCore/Repositories/ResourceRepositoryAccessor.cs:line 47
   at async Task<IReadOnlyCollection<TResource>> JsonApiDotNetCore.Services.JsonApiResourceService<TResource, TId>.GetAsync(CancellationToken cancellationToken) in C:/projects/jsonapidotnetcore/src/JsonApiDotNetCore/Services/JsonApiResourceService.cs:line 67
   at async Task<IActionResult> JsonApiDotNetCore.Controllers.BaseJsonApiController<TResource, TId>.GetAsync(CancellationToken cancellationToken) in C:/projects/jsonapidotnetcore/src/JsonApiDotNetCore/Controllers/BaseJsonApiController.cs:line 100
   at async Task<IActionResult> JsonApiDotNetCore.Controllers.JsonApiController<TResource, TId>.GetAsync(CancellationToken cancellationToken) in C:/projects/jsonapidotnetcore/src/JsonApiDotNetCore/Controllers/JsonApiController.cs:line 51
   at async ValueTask<IActionResult> Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor+TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, object controller, object[] arguments)
   at TResult System.Runtime.CompilerServices.ValueTaskAwaiter<TResult>.GetResult()
   at async Task Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()+Awaited(?)
   at async Task Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()+Awaited(?)
   at void Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Task Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
   at Task Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
   at async Task Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeNextExceptionFilterAsync()+Awaited(?)
```

Reply:
`new` breaks inheritance, as if the property exists twice. Did you try 'override' instead?

Reply:
I hadn't tried it, no, but I ran into the same error when I did :(
Also, this added another issue, I personally do not want to have MongoDb persist the `StringId` field but I cannot `override` it in order to add the `[BsonIgnore]` annotation because it ain't `virtual`.